lib/game.rb

```diff
@@ -0,0 +1,94 @@
+class Game
+	attr_accessor :board, :player_1, :player_2
+	WIN_COMBINATIONS = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]]
+
+
+	def initialize(player_1 = Player::Human.new("X"), player_2 = Player::Human.new("O"), board = Board.new)
+		@board = board
+		@player_1 = player_1
+		@player_2 = player_2
+		
+	end
+
+	def current_player
+		if self.board.turn_count.odd?
+			return self.player_2
+		else
+			return self.player_1
+		end
+	end
+
+	def over?
+		board.full? || self.won? || self.draw?
+	end
+
+	def won?
+			a = []
+			WIN_COMBINATIONS.each do |i|
+				if i.all?{|item| self.board.cells[item] == "X"} || i.all?{|item| self.board.cells[item] == "O"}
+				a << i
+				end
+			end 
+			if a == []
+				false
+			else 
+				true
+			end 
+	end
+
+	def draw?
+		self.board.full? && !self.won?
+	end
+
+	def winner
+		a = board.cells.select{|item| item == "X"}
+		b = board.cells.select{|item| item == "O"}
+		if self.won?
+			if a.length > b.length
+				return "X"
+		    else
+			    return "O"
+			end
+		else
+			nil
```

Comment:
You do not need to explicitly return `nil` from an `if` statement; an `if` will return `nil` if the condition fails.
```
if self.won?
  if a.length > b.length
    return "X"
  else
    return "O"
  end
end
```